apm-agent-plugins/apm-awslambda-plugin/src/main/java/co/elastic/apm/agent/awslambda/helper/AbstractLambdaTransactionHelper.java

```diff
@@ -143,6 +143,8 @@ private void completeMetaData(Context lambdaContext) {
             String region = arnSegments[3];
             String accountId = arnSegments[4];
 
+            String logStreamName = lambdaContext.getLogStreamName();
```

Comment:
unfortunately, after snapshot restore, this method returns null, so lambda instance has \<uknown\> node name instead of CloudWatch log stream name.

AWS team replied, that this environment variable won't be available, since it's impossible to change env variables after snapshot it taken.

We need to decide for a fallback strategy, how to generate lambda instance identifier.

Reply:
I've got a reply from AWS that neither AWS_LAMBDA_LOG_STREAM_NAME, neither context.getLogStreamName gonna be populated for SnapStart enabled functions - https://github.com/aws/aws-lambda-java-libs/issues/402 .

I do not know, if Kibana product have some special handling of AWS lambda APM data and if it relies on the format of configured_name from service.node field.

In initial proposal, I suggest to simulate and generate similar to log group node name, but right now I think we do not need to tie (if possible) instance name to log group name at all.

If we need just to distinguish different lambda instances, may be pair of lambda version + ephemeral id is enough, but this is a breaking change, if we are going to change it for all AWS Lambdas.

Another option I'm considering is to introduce a property, which will be a strategy for lambda instance naming.
By default, we should still try to use AWS_LAMBDA_LOG_STREAM_NAME env variable, but we may allow another one, which will use auto generated name.